### ISSUE TITLE
Investigate cause of "Converted or Backfilled rating" error from BGS

### ISSUE THREAD

**Issue by leikkisa** [department-of-veterans-affairs/caseflow#7054]
Handle the situation where we get the below response when looking for timely_ratings from BGS.

```{:error_level=>"0", :any_ratings_within_range_flag=>"N", :differences_found=>"N", :multiple_ratings_flag=>"N", :ptcpnt_vet_id=>"123456", :reject_reason=>"Converted or Backfilled Rating - no promulgated ratings found", :rating_profile_list=>nil}```

When a Veteran has no ratings at all, we're still getting the same Savon error as before.  Some of the Veteran's are also returning a "locked rating" error (which is in another ticket).

**Comment by leikkisa** [comment]
Another response we can get that causes a similar problem is:
{:error_level=>"0", :reject_reason=>"Locked Rating Detected "}

**Comment by leikkisa** [comment]
This seems to be affecting Veterans which we know have ratings.  I checked a Veteran that does not have ratings, and we're getting the same Savon error as before.  If a Veteran with ratings isn't getting this error, they're getting the locked rating error.  I haven't been able to successfully get ratings for any Veteran since this issue started.

I have asked VBMS and they're going to ask their ratings person tomorrow.

**Comment by shanear** [comment]
This is happening in prod and preventing work, bumping to high priority

**Comment by shanear** [comment]
This is now no longer happening in prod?? We should investigate why this happens.

**Comment by leikkisa** [comment]
After talking to BGS, we heard that this should only happen for really old ratings (like 10 years old), but BGS also mentioned that it is happening for newer ratings.  They said that we do not want these ratings to show up.  This shouldn't block us from getting other ratings, but we need to confirm that this is the case.

It seems like this might be caused by improper test data set up.

It is somewhat a mystery to me why it was happening on some Veterans and is no longer happening for them (but is happening for new test data that has been set up), but will not worry about that until it happens again.

The next step is to send BGS of some examples of this happening.

**Comment by leikkisa** [comment]
Test file numbers have been sent:
700032764
700032765
700032767
700032768
700032770
700032771
700032773
700032774

**Comment by leikkisa** [comment]
If we get this error,  it means the ratings have to be part of an award and one hasn't yet been generated and authorized.

We still don't understand why some that have worked before broke, and other mysteries, but I am going to close this for now. It seems like this error should still cause an error in Caseflow, and if we need to fix it, we can have someone go into VBMS-A and generate and authorize an award for the rating.

**Comment by kencheeto** [comment]
We started seeing this error again today: https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/2967/

Here's my current understanding of what's going on after chatting a bit with Sally:

Some veterans have a "backfilled or converted" rating in BGS, which results in a `NilRatingProfileListError` when we try to fetch ratings when creating an intake or displaying an existing intake. This means that

1. when an intake is started for this veteran, the created intake fails to display and the user sees an error page
1. when the user visits Caseflow later, the intake is still in progress and Caseflow displays an error page. The user is unable to cancel the intake.
1. when the user tries to edit an intake for a veteran with a backfilled/converted rating, Caseflow shows an error page

It's unclear when a rating can become backfilled or converted. Anecdotally Sally mentioned that it might be by accident - some were accidentally created in the past in UAT, so it could be that the same thing happened in prod. One fix that would unblock Caseflow Intake users would be to cancel the intake-in-progress for the CA, then notify them that this veteran cannot be intaked until we can figure out how to fix (or ask someone to fix) the backfill/converted rating.

### Do backfilled/converted ratings prevent other valid ratings from being returned? 

"backfilled or converted" means the rating has not been promulgated, so it can be considered an incomplete rating (and cannot be intaked again). However, one problem we see with locked ratings is that any locked rating for a veteran stops BGS from returning any ratings at all for the vet (so one locked rating and 2 promulgated ratings would mean BGS returns nothing). Sally has asked Colin to help us setup test data to verify this theory.

**Comment by leikkisa** [comment]
We are going to have a meeting with Linda and Colin next week. Colin says that backfilled ratings should be treated like regular ratings, and we should be able to process them.  However, most real backfilled ratings should be really old (so not timely).  According to Linda, if a recent rating is backfilled, it has a different meaning and it is not promulgated yet, and shouldn't be included in the ratings.

I looked up the Veteran on prod that is experiencing this issue, and I get the error no matter what date range I put in (even 1 day), and do not get any promulgated ratings. So I think that the time range we put in isn't relevant to the error.

**Comment by leikkisa** [comment]
Backfilled or converted ratings could possibly be a rating that could be eligible for a Decision Review. Formerly denials did not get promulgated.  If a change is made to an award, that does not get promulgated.  There is a legal question as to whether we should show these, if they do not have a legal document attached.

If there are any promulgated ratings, they should be returned and they shouldn't get this error. We believe this error should only happen if there are no promulgated ratings and there is a Backfilled or Converted rating.

We've decided to move forward at the moment by letting the user proceed as if there are no promulgated ratings, and having them enter their issue with freetext using the unidentified issue option. 

We still want to track when this happens so that Linda can have more cases to investigate (we currently only have one prod case), which was from Dec 2017, so it can happen with ratings that are not from the old system as previously thought.

**Comment by leikkisa** [comment]
Closing the investigation ticket. New development ticket opened here: https://github.com/department-of-veterans-affairs/caseflow/issues/8494
